### PR TITLE
[basic-auth] Add auth.parse

### PR DESCRIPTION
**Pull request by vesse**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jshttp/basic-auth/blob/v1.1.0/index.js#L87-L95
- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by aozgaa** [comment]
Can you copy the jsdoc comment for `parse` instead of including the single-line comment? We use the jsdoc comment when providing intellisense.

**Comment by vesse** [comment]
@aozgaa Sure, and done.

**Comment by vesse** [comment]
@aozgaa Any change getting this merged?

**Comment by vesse** [comment]
Since the first reviewer unsubscribed himself from reviews in #16541 maybe someone else could look into this? Mentioning @alloy just because his merge commit is the latest commit.

**Comment by alloy** [comment]
Sorry, I know nothing of these rulings.

Paging @andy-ms 🙏🏽

File: types/basic-auth/index.d.ts [hunk 1]
```diff
@@ -14,6 +14,15 @@ declare namespace auth {
         name: string;
         pass: string;
     }
+
+    /**
+     * Parse basic auth to object.
+     *
+     * @param {string} string
```
Comment (andy-ms) [inline types/basic-auth/index.d.ts]: JSDoc is redundant here, besides the first line.

**Comment by vesse** [comment]
@andy-ms Removed other than the first line

**Pull request merged**